src/fxa-geodb.js

```diff
@@ -0,0 +1,79 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const db = '../db/cities-db.mmdb';
+const db_backup = '../db/cities-db.mmdb-backup';
+const ERRORS = require('../lib/errors');
+const Joi = require('joi');
+const maxmind = require('maxmind');
+const Promise = require('bluebird');
+
+function GeoDB(ip, time_stamp) {
+  'use strict';
+  time_stamp = time_stamp || new Date();
+
+  // allows us to check whether the ip is defined, is a string, and is not empty
+  var schema = Joi.object().keys({
+    IP: Joi.string().required()
+  });
+  var err = Joi.validate({IP: ip}, schema);
+  if (err.error) {
+    return Promise.reject({
+      message: err.error.details[0].message
+    });
+  }
+
+  // at this point, we know ip is a string and is
+  // non-empty, we can validate it
+  if (! maxmind.validate(ip)) {
```

Comment:
from above `the ip is defined, is a string, and is not empty` does `maxmind.validate(` not do that for us already?